specs/rollup-node.md

```diff
@@ -0,0 +1,431 @@
+    - Batches not matching filter criteria are ignored:
+      - `batch.epoch == sequencing_window.epoch`, i.e. for this sequencing window
+      - `(batch.timestamp - genesis_l2_timestamp) % block_time == 0`, i.e. timestamp is aligned
+      - `min_l2_timestamp <= batch.timestamp < max_l2_timestamp`, i.e. timestamp is within range
+        - `min_l2_timestamp = prev_l2_timestamp + l2_block_time`
+          - `prev_l2_timestamp` is the timestamp of the previous L2 block: the last block of the previous epoch,
+              or the L2 genesis block timestamp if there is no previous epoch.
+          - `l2_block_time` is a configurable parameter of the time between L2 blocks
+        - `max_l2_timestamp = max(l1_timestamp + max_sequencer_drift, min_l2_timestamp + l2_block_time)`
+          - `l1_timestamp` is the timestamp of the L1 block associated with the L2 block's epoch
+          - `max_sequencer_drift` is the most a sequencer is allowed to get ahead of L1
+      - The batch is the first batch with `batch.timestamp` in this sequencing window,
+        i.e. one batch per L2 block number.
+      - The batch only contains sequenced transactions, i.e. it must NOT contain any Deposit-type transactions.
+
+Note that after the above filtering `min_l2_timestamp >= l1_timestamp` always holds,
+i.e. a L2 block timestamp is always equal or ahead of the timestamp of the corresponding L1 origin block.
+
+[random]: https://eips.ethereum.org/EIPS/eip-4399
+
+A bundle of batches is versioned by prefixing with a bundle version byte: `bundle = bundle_version ++ bundle_data`.
+
+Bundle versions:
+
+- `0`: `bundle_data = RLP([batch_0, batch_1, ..., batch_N])`
+- `1`: `bundle_data = compress(RLP([batch_0, batch_1, ..., batch_N]))` (compression algorithm TBD)
+
+A batch is also versioned by prefixing with a version byte: `batch = batch_version ++ batch_data`
+and encoded as a byte-string (including version prefix byte) in the bundle RLP list.
+
+Batch versions:
+
+- `0`: `batch_data = RLP([epoch, timestamp, transaction_list])`, where each
+
+Batch contents:
+
+- `epoch` is the sequencing window epoch, i.e. the first L1 block number
+- `timestamp` is the L2 timestamp of the block
+- `transaction_list` is an RLP encoded list of [EIP-2718] encoded transactions.
+
+[EIP-2718]: https://eips.ethereum.org/EIPS/eip-2718
+
+The L1 attributes are read from the L1 block header, while deposits are read from the block's [receipts][g-receipts].
+Refer to the [**deposit contract specification**][deposit-contract-spec] for details on how deposits are encoded as log
+entries. The deposited and sequenced transactions are combined when the Payload Attributes are constructed.
+
+[deposit-contract-spec]: deposits.md#deposit-contract
+
+### Encoding the L1 Attributes Deposited Transaction
+
+The [L1 attributes deposited transaction][g-l1-attr-deposit] is a call that submits the L1 block attributes (listed
+above) to the [L1 attributes predeployed contract][g-l1-attr-predeploy].
```

Comment:
```suggestion
The [L1 attributes deposited transaction][g-l1-attr-deposit] is a call (occurring on L2)  that submits the L1 block attributes (listed
above) to the [L1 attributes predeployed contract][g-l1-attr-predeploy].
```

I feel like the term L1 appears so much in this sentence that it should be clarified where this is happening.